src/main/java/jcifs/smb1/UniAddress.java

```diff
@@ -0,0 +1,470 @@
+                    tmp[i++] = RESOLVER_LMHOSTS;
+                } else if( s.equalsIgnoreCase( "WINS" )) {
+                    if( nbns == null ) {
+                        if( log.level > 1 ) {
+                            log.println( "UniAddress resolveOrder specifies WINS however the " +
+                                    "jcifs.smb1.netbios.wins property has not been set" );
+                        }
+                        continue;
+                    }
+                    tmp[i++] = RESOLVER_WINS;
+                } else if( s.equalsIgnoreCase( "BCAST" )) {
+                    tmp[i++] = RESOLVER_BCAST;
+                } else if( s.equalsIgnoreCase( "DNS" )) {
+                    tmp[i++] = RESOLVER_DNS;
+                } else if( log.level > 1 ) {
+                    log.println( "unknown resolver method: " + s );
+                }
+            }
+            resolveOrder = new int[i];
+            System.arraycopy( tmp, 0, resolveOrder, 0, i );
+        }
+    }
+
+    static class Sem {
+        Sem( int count ) {
+            this.count = count;
+        }
+        int count;
+    }
+
+    static class QueryThread extends Thread {
+    
+        Sem sem;
+        String host, scope;
+        int type;
+        NbtAddress ans = null;
+        InetAddress svr;
+        UnknownHostException uhe;
+    
+        QueryThread( Sem sem, String host, int type,
+                        String scope, InetAddress svr ) {
+            super( "JCIFS-QueryThread: " + host );
+            this.sem = sem;
+            this.host = host;
+            this.type = type;
+            this.scope = scope;
+            this.svr = svr;
+        }
+        public void run() {
+            try {
+                ans = NbtAddress.getByName( host, type, scope, svr );
+            } catch( UnknownHostException uhe ) {
+                this.uhe = uhe;
+            } catch( Exception ex ) {
+                this.uhe = new UnknownHostException( ex.getMessage() );
+            } finally {
+                synchronized( sem ) {
+                    sem.count--;
+                    sem.notify();
+                }
+            }
+        }
+    }
+
+    static NbtAddress lookupServerOrWorkgroup( String name, InetAddress svr )
+                                                    throws UnknownHostException {
+        Sem sem = new Sem( 2 );
+        int type = NbtAddress.isWINS( svr ) ? 0x1b : 0x1d;
+
+        QueryThread q1x = new QueryThread( sem, name, type, null, svr );
+        QueryThread q20 = new QueryThread( sem, name, 0x20, null, svr );
+        q1x.setDaemon( true );
+        q20.setDaemon( true );
+        try {
+            synchronized( sem ) {
+                q1x.start();
+                q20.start();
+
+                while( sem.count > 0 && q1x.ans == null && q20.ans == null ) {
+                    sem.wait();
+                }
+            }
+        } catch( InterruptedException ie ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)